report/topology.go

```diff
@@ -195,6 +197,100 @@ func (c Counters) Copy() Counters {
 	return result
 }
 
+// Sets is a string->set-of-strings map.
+type Sets map[string]StringSet
+
+// Merge merges two sets maps into a fresh set, performing set-union merges as
+// appropriate.
+func (s Sets) Merge(other Sets) Sets {
+	result := s.Copy()
+	for k, v := range other {
+		result[k].Merge(v)
+	}
+	return result
+}
+
+// Copy returns a value copy of the sets map.
+func (s Sets) Copy() Sets {
+	result := Sets{}
+	for k, v := range s {
+		result[k] = v.Copy()
+	}
+	return result
+}
+
+// StringSet is a sorted set of unique strings. Clients must use the Add
+// method to add strings.
```

Comment:
Okay I had a stab at that but its not very clean - it would seem the type aliasing doesn't propagate methods.  I tried embedding it in a struct too - that didn't work either.

https://github.com/weaveworks/scope/commit/b867b0465a252a06329df4ed199990d8fd757bb5

Thoughts?